core/src/main/java/jenkins/model/queue/AsynchronousExecution.java

```diff
@@ -106,7 +106,11 @@ public synchronized final Executor getExecutor() {
         return executor;
     }
 
-    @Restricted(NoExternalUse.class)
+    /**
+     * Use {@link #setExecutorWithoutCompleting(Executor)} and {@link #maybeComplete()}
+     * instead. For this method, locking behaviour is problematic.
+     */
+    @Restricted(NoExternalUse.class) @Deprecated()
```

Comment:
Likely it's safe to change the annotation to `DoNotUse.class`. It has been already restricted, and now you really do not want it to be used in the core as well.

Reply:
Done.